app/src/main/java/com/konarsubhojit/synckro/domain/sync/SyncDiffer.kt

```
@@ -0,0 +1,169 @@
+        val localNewerThanRemote: Boolean,
+    ) : SyncOp
+}
+
+/**
+ * Pure function from (local state, remote state, last-known index) to a list
+ * of operations. Has no Android or IO dependencies, and is the unit of logic
+ * covered by unit tests.
+ */
+object SyncDiffer {
+
+    fun diff(
```

detekt (1.23.8)

[warning] The function diff appears to be too complex based on Cyclomatic Complexity (complexity: 49). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)